src/main/java/io/stargate/sgv2/jsonapi/service/operation/model/impl/DeleteOperation.java

```diff
@@ -27,7 +29,8 @@ public record DeleteOperation(
      * Added parameter to pass number of document to be deleted, this is needed because read
      * documents limit changed to deleteLimit + 1
      */
-    int deleteLimit)
+    int deleteLimit,
+    int retryLimit)
     implements ModifyOperation {
   @Override
   public Uni<Supplier<CommandResult>> execute(QueryExecutor queryExecutor) {
```

Comment:
Would be great if you can start adding some improvements to readbility of reactive seq creation:

* add inline comments that clearly split the steps, like here f.e.: https://github.com/stargate/jsonapi/blob/46aa965df35ff6ec7f520712298ae0a2abbf4d17/src/main/java/io/stargate/sgv2/jsonapi/service/operation/model/impl/InsertOperation.java#L46-L94
* avoid doing variable assignment..for example:

```
FROM

  private Uni<? extends ReadDocument> readDocumentAgain(
      QueryExecutor queryExecutor, ReadDocument prevReadDoc) {
    // Read again if retry flag is `true`
    final Uni<ReadOperation.FindResponse> findResponse =
        readOperation()
            .getDocuments(
                queryExecutor,
                null,
                new DBFilterBase.IDFilter(DBFilterBase.IDFilter.Operator.EQ, prevReadDoc.id()));
    return findResponse
        .onItem()
        .transform(
            response -> {
              if (!response.docs().isEmpty()) {
                return response.docs().get(0);
              } else {
                // If data changed and doesn't satisfy filter conditions
                return null;
              }
            });
  }

TO

  private Uni<? extends ReadDocument> readDocumentAgain(
      QueryExecutor queryExecutor, ReadDocument prevReadDoc) {
    // Read again if retry flag is `true`
    return readOperation()
            .getDocuments(
                queryExecutor,
                null,
                new DBFilterBase.IDFilter(DBFilterBase.IDFilter.Operator.EQ, prevReadDoc.id()))

        // then map
        .onItem()
        .transform(
            response -> {
              if (!response.docs().isEmpty()) {
                return response.docs().get(0);
              } else {
                // If data changed and doesn't satisfy filter conditions
                return null;
              }
            });
  }

```

Reply:
Ok will do